### PR TITLE
uuid.c: fix checking valid chars

### PR DESCRIPTION
**Pull request by kemonats**
## Description

Changes:
- possibility use of uppercases
- characters from 0x3a to 0x3f were eliminated as valid
- added checking for valid group delimiter
- first hex character validation added
- the condition for using the uuid_useRand () function has been changed

Values returned by the uuid_unpare () function:
```
uuid                                 desc                        now  after
AA000000-0000-0000-0000-000000000000 uppercases                  -1    0
00000000=0000-0000-0000-000000000000 '=' as separator             0   -1
w0000000-0000-0000-0000-000000000000 first char bad, second good  0   -1
==000000-0000-0000-0000-000000000000 '=' as char                  0   -1
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: libuuid/uuid.c [hunk 1]
```diff
@@ -87,31 +101,35 @@ void uuid_unparse(const uuid_t uu, char *out)
 
 int uuid_parse(const char *in, uuid_t uu)
 {
-	int i, ret;
+	int ret;
 	unsigned char val;
-	const char *temp;
```
Comment (agkaminski) [inline libuuid/uuid.c:92]: Modification of function parameters is by some considered a bad practice

**Review by agkaminski — APPROVED** [review]

File: libuuid/uuid.c (hunk 1 above)
Comment (damianloew) [reply libuuid/uuid.c:92]: I will opt for leaving the temp variable and not changing the `in` paremeter

File: libuuid/uuid.c [hunk 2]
```diff
@@ -158,7 +176,8 @@ void uuid_generate_random(uuid_t out)
 
 		fclose(file);
 	}
-	else if (useRand > 0) {
+
+	if (useRand > 0) {
```
Comment (damianloew) [inline libuuid/uuid.c:180]: I think this change is unnecessary here

**Review by damianloew — CHANGES_REQUESTED** [review]
Thanks for the corrections, I tested the changes using UUID unit tests and they have passed. There are only small remarks from me

File: libuuid/uuid.c (hunk 1 above)
Comment (kemonats) [reply libuuid/uuid.c:92]: @agkaminski But you use it e.g. here [string.c](https://github.com/phoenix-rtos/libphoenix/blob/master/string/string.c).

**Review by damianloew — APPROVED** [review]

**Pull request merged**